src/OpenRoad.cc

```diff
@@ -390,17 +390,26 @@ void OpenRoad::writeLef(const char* filename)
       if (cnt > 0) {
         name += "_" + std::to_string(cnt);
         std::ofstream os(name);
+        if (!os.is_open()) {
```

Comment:
Exceptions are enabled when reading. The advantage is that *all* IO operations are checked, including each individual read/write operation, so if one runs out of e.g. disk space, an exception is thrown.

https://github.com/The-OpenROAD-Project/OpenROAD/blob/7586d03811ca43cc2615077afb0f4821c81f82e2/src/OpenRoad.cc#L426

The code below has is manually adding the exceptions.

Reply:
@maliberty Do we correctly catch these exceptions in TCL land?

Reply:
https://github.com/The-OpenROAD-Project/OpenROAD/blob/7586d03811ca43cc2615077afb0f4821c81f82e2/src/Exception.i#L28

should catch them but you can always verify